File: 현호/BOJ_10773.js
// 백준 10773 : 제로
// 통과여부 : O
// 설명 : 최종적으로 적어낸 수의 합을 출력하기

const fs = require("fs");
const input = fs
    .readFileSync("/dev/stdin")
    .toString()
    .trim()
    .split("\n")
    .map(Number);
const command_amount = input.shift();

const solve = () => {
    const arr = [];
    let sum = 0;

    for (let i = 0; i < command_amount; i += 1) {
        if (input[i]) arr.push(input[i]);
        else arr.pop();
    }

    arr.forEach((el) => (sum += el));
    return sum;
}

console.log(solve());

File: 근재/BOJ_10773.py
# 돈을 잘못 부를 때마다 0을 외쳐서, 가장 최근에 재민이가 쓴 수를 지우게 시킴
# 이렇게 모든 수를 받아 적은 후 그 수의 합을 알고 싶다.
K = int(input())
# 값들을 저장할 Array
arr = []

for i in range(K):
    # 정수 입력
    N = int(input())
    # 0을 만나면 맨 위의 인덱스값 pop
    if N == 0:
        arr.pop()
    else:
    # 아닌 경우에는 값을 저장
        arr.append(N)

print(sum(arr))